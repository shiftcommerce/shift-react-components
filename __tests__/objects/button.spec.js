// Libraries
import React from 'react'

// Object
import Button from '../../src/objects/button'

test('renders correctly', () => {
  // Arrange & Act
  const wrapper = mount(
    <Button label='ADD TO BAG' status='positive' size='lrg' />
  )

  // Assert
  expect(wrapper).toMatchSnapshot()
})

test('disabled button', () => {
  // Arrange & Act
  const wrapper = mount(
    <Button label='Testing Button' disabled size='lrg' status='primary' />
  )

  // Assert
  expect(wrapper).toMatchSnapshot()
  expect(wrapper.find('button')).toHaveClassName('o-button--disabled')
  expect(wrapper.find('button')).toHaveProp('disabled', true)
})

describe('Button by default', () => {
  test('disabled should be false', () => {
    // Arrange & Act
    const wrapper = mount(
      <Button label='Enabled Button' size='lrg' status='primary' />
    )

    // Assert
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('button')).not.toHaveClassName('o-button--disabled')
  })
})

describe('Button with props', () => {
  test('should be of large size', () => {
    // Arrange & Act
    const wrapper = mount(
      <Button label='Large Button' size='lrg' />
    )

    // Assert
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('button')).toHaveClassName('o-button--lrg')
  })

  test('status should be primary', () => {
    // Arrange & Act
    const wrapper = mount(
      <Button label='Primary Button' status='primary' />
    )

    // Assert
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('button')).toHaveClassName('o-button--primary')
  })
})
