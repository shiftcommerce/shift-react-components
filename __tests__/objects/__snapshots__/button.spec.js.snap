// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button by default disabled should be false 1`] = `
<Button
  label="Enabled Button"
  size="lrg"
  status="primary"
>
  <button
    className="o-button o-button--primary o-button--lrg"
    disabled={false}
    role="button"
    tabIndex="0"
  >
    Enabled Button
  </button>
</Button>
`;

exports[`Button with props should be of large size 1`] = `
<Button
  label="Large Button"
  size="lrg"
>
  <button
    className="o-button o-button--lrg"
    disabled={false}
    role="button"
    tabIndex="0"
  >
    Large Button
  </button>
</Button>
`;

exports[`Button with props status should be primary 1`] = `
<Button
  label="Primary Button"
  status="primary"
>
  <button
    className="o-button o-button--primary"
    disabled={false}
    role="button"
    tabIndex="0"
  >
    Primary Button
  </button>
</Button>
`;

exports[`disabled button 1`] = `
<Button
  disabled={true}
  label="Testing Button"
  size="lrg"
  status="primary"
>
  <button
    className="o-button o-button--primary o-button--lrg o-button--disabled"
    disabled={true}
    role="button"
    tabIndex="0"
  >
    Testing Button
  </button>
</Button>
`;

exports[`renders correctly 1`] = `
<Button
  label="ADD TO BAG"
  size="lrg"
  status="positive"
>
  <button
    className="o-button o-button--positive o-button--lrg"
    disabled={false}
    role="button"
    tabIndex="0"
  >
    ADD TO BAG
  </button>
</Button>
`;
